routes.py

```diff
@@ -0,0 +1,663 @@
+        elif len(user_role_in_team) > 0:
+            flash("This user's profile is already a member of this team", 'error')
+        else:
+            new_inv = ClashInvitation(
+                futureteam_id=inv_for_team.id,
+                role=form.role.data,
+                lider_id=user.id,
+                guestprofile_id=inv_guest.id,
+            )
+            db.session.add(new_inv)
+            db.session.commit()
+            flash('Your invitation has been sent', 'success')
+            return redirect(url_for('findteammates'))
+    return redirect(url_for('findteammates'))
+
+
+# ścieżka odpowiedzi na wpływające zaproszenia,
+@app.route('/user/<username>/mailbox/invitations/answer', methods=['GET', 'POST'])
+@login_required
+def inv_answer(username):
+    user = User.query.filter_by(username=username).first_or_404()
+    form = AnswerForm()
+    if form.validate_on_submit():
+        in_invitation_id = request.form['id_in']
+        reviewd_invitation = ClashInvitation.query.filter_by(
+            id=in_invitation_id).first_or_404()
+        if form.reject_button.data:
+            db.session.delete(reviewd_invitation)
+            db.session.commit()
+            flash('The invitation has been rejected.', 'message')
+        elif form.accept_button.data:
+            team_to_join = ClashTeam.query.filter_by(
+                id=reviewd_invitation.futureteam_id).first_or_404()
+            join_position = getattr(
+                team_to_join, reviewd_invitation.role.lower())
+            new_teammember = Profile.query.filter_by(
+                id=reviewd_invitation.guestprofile_id).first_or_404()
+            if join_position is not None:
+                flash('This position is already taken', 'error')
+            else:
+                if reviewd_invitation.role == 'Toplane':
+                    team_to_join.add_top(new_teammember.nickname)
+                elif reviewd_invitation.role == 'Jungle':
+                    team_to_join.add_jungle(new_teammember.nickname)
+                elif reviewd_invitation.role == 'Midlane':
+                    team_to_join.add_midlane(new_teammember.nickname)
+                elif reviewd_invitation.role == 'Bottom':
+                    team_to_join.add_bottom(new_teammember.nickname)
+                else:
+                    team_to_join.add_support(new_teammember.nickname)
+                flash('You have joined the team successfully.', 'success')
+                db.session.delete(reviewd_invitation)
+                db.session.add(team_to_join)
+                db.session.commit()
+    return redirect(url_for('mailbox', username=user.username))
+
+
+# ścieżka wycofywania wysłanych zaproszeń,
+@app.route('/user/<username>/mailbox/invitations/answer/withdrawal', methods=['GET', 'POST'])
+@login_required
+def inv_withdrawal(username):
+    user = User.query.filter_by(username=username).first_or_404()
+    form = AnswerForm()
+    if form.validate_on_submit():
```

Comment:
There is this line in a lot of functions. I **think** that's because you are handling both methods here - GET and POST. Generally, you're going to have submitted data coming through POST, so you can split the handling into two separate functions (GET displaying the form, POST processing the data).